File: Go/casino/player.go
package casino

type Player struct {
	isInGame bool
}

func (self *Player) IsIn(game *Game) bool {
	return self.isInGame
}

func (self *Player) Join(game *Game) error {
	if err := game.AddPlayer(); err != nil {
		return err
	}

	self.isInGame = true
	return nil
}

func (self *Player) Leave() {
	self.isInGame = false
}

File: NodeJS/src/Player.ts
import { RollDiceGame } from "./RollDiceGame";

export class Player {
    _isInGame = false;

    enter(game: RollDiceGame) {
        if (this._isInGame) {
            throw new TypeError("Unable to enter another game");
        }
        this._isInGame = true;
    }

    leaveGame() {
        if (!this._isInGame) {
            throw new TypeError("Unable to leave the game before entering");
        }
        this._isInGame = false;
    }

    isInGame(): boolean {
        return this._isInGame
    }
}